pkgs/build-support/rust/import-cargo-lock.nix

```diff
@@ -33,7 +33,7 @@ let
     then builtins.readFile lockFile
     else args.lockFileContents;
 
-  packages = (builtins.fromTOML lockFileContents).package;
+  packages = (builtins.fromTOML (builtins.unsafeDiscardStringContext lockFileContents)).package;
```

Comment:
I think any time we use `builtins.unsafe*`, that at least warrants a comment explaining why that use is safe.

Reply:
added a note. though this does raise the question of whether the JSON and TOML readers shouldn't just propagate context as well?

Reply:
it cannot propagate context if the resulting value does not contain a string, eg `fromJSON '{ "some_hash": 2 }'
But I wonder if builtins.readFile should only carry the context of the content of the file, not of the full store path.

Reply:
Related: https://github.com/NixOS/nix/issues/6647